### PR TITLE
Add 'Remove Defect(s)' feature for Dexter Admin.

### PR DESCRIPTION
**Pull request by Minjung-Baek**
remove_defect_image

### CODE REVIEW

**Comment by Minjung-Baek** [comment]
remove_defect_image

File: project/dexter-server/public/AdminSE/adminSECtrl.js [hunk 1]
```diff
@@ -1,101 +1,200 @@
...
         removeModulePathList(event);
         changeSelectedModuleCount();
     });
 
-    function changeSelectedModuleCount(){
-        var count = $scope.selectedModulePathList.length;
-        $("#removeModuleCount").html(parseInt(count));
-        if(count > 0) {
+    function changeSelectedModuleCount() {
+        const count = parseInt($scope.selectedModulePathList.length);
```
Comment (mdopark) [inline project/dexter-server/public/AdminSE/adminSECtrl.js]: @Minjung-Baek Array.length is always a integer. parseInt() is not necessary.

File: project/dexter-server/public/AdminSE/adminSECtrl.js [hunk 2]
```diff
@@ -1,101 +1,200 @@
...
-        }else{
+        } else {
             $("#isModuleSelected").removeClass("fa fa-folder-open").addClass("fa fa-folder");
         }
     }
 
-    function addModulePathList(event){
-        if (!event) { return; }
+    function changeSelectedDidCount() {
+        const count = parseInt($scope.selectedDidList.length);
```
Comment (mdopark) [inline project/dexter-server/public/AdminSE/adminSECtrl.js]: Array.length is always a integer. parseInt() is not necessary.

File: project/dexter-server/public/AdminSE/adminSECtrl.js [hunk 3]
```diff
@@ -1,101 +1,200 @@
...
+                $scope.didList = [];
+                angular.forEach(result.data.rows, (index) => {
+                    $scope.didList.push(index.did);
+                });
+
+                $(".did-select").select2({
+                    data: $scope.didList
+                });
+                hideLoadingImage();
+            }
```
Comment (mdopark) [inline project/dexter-server/public/AdminSE/adminSECtrl.js]: 126, 128 lines : LoadingImage shoud be hide on error.

File: project/dexter-server/public/AdminSE/adminSECtrl.js [hunk 4]
```diff
@@ -1,101 +1,200 @@
...
                     $scope.modulePathList.push(index.modulePath);
                 });
                 $scope.modulePathList.push(NONE_MODULE_PATH);
 
-                $(".select2").select2({
-                    data : $scope.modulePathList
+                $(".module-select").select2({
+                    data: $scope.modulePathList
                 });
+                hideLoadingImage();
```
Comment (mdopark) [inline project/dexter-server/public/AdminSE/adminSECtrl.js]: It was called already at 138 line.

File: project/dexter-server/public/AdminSE/adminSECtrl.js [hunk 5]
```diff
@@ -1,101 +1,200 @@
...
             }
-        }, function(results){
-            $log.error('Error code:' + results.status+';');
+        }, function (results) {
+            $log.error('Error code:' + results.status + ';');
```
Comment (mdopark) [inline project/dexter-server/public/AdminSE/adminSECtrl.js]: 149, 151 line : LoadingImage shoud be hide on error.

File: project/dexter-server/public/AdminSE/adminSECtrl.js [hunk 6]
```diff
@@ -1,101 +1,200 @@
...
             }
-        }).then(function(result){
-            if(isHttpResultOK(result)){
-                isHiddenLoadingImage();
+        }).then((result) => {
+            if (isHttpResultOK(result)) {
+                hideLoadingImage();
                 alert("Your removal request is completed.");
                 location.reload(true);
             }
```
Comment (mdopark) [inline project/dexter-server/public/AdminSE/adminSECtrl.js]: LoadingImage shoud be hide when the http result is not OK.

File: project/dexter-server/public/AdminSE/adminSECtrl.js [hunk 7]
```diff
@@ -1,101 +1,200 @@
...
     }
 
-    var init = function(){
-        isShownLoadingImage();
+    function addDidList(event) {
+        if (!event) {
+            return;
+        }
+
+        JSON.stringify(event.params, function (key, value) {
```
Comment (mdopark) [inline project/dexter-server/public/AdminSE/adminSECtrl.js]: @Minjung-Baek I think JSON.stringify() is not suitable for just looping.
Is not forEach() better ?

File: project/dexter-server/public/AdminSE/adminSECtrl.js [hunk 8]
```diff
@@ -43,42 +43,31 @@ adminSEApp.controller('adminSECtrl', function ($scope, $http, $location, $routeP
...
     function addModulePathList(event) {
-        if (!event) {
+        if (!event)
             return;
-        }
-        JSON.stringify(event.params, function (key, value) {
-            $scope.selectedModulePathList.push(value.data.id);
-        });
+
+        $scope.selectedModulePathList.push(event.params.data.id);
```
Comment (mdopark) [inline project/dexter-server/public/AdminSE/adminSECtrl.js]: Not need to loop ??

File: project/dexter-server/public/AdminSE/adminSECtrl.js [hunk 9]
```diff
@@ -43,42 +43,31 @@ adminSEApp.controller('adminSECtrl', function ($scope, $http, $location, $routeP
...
     function removeModulePathList(event) {
-        if (!event) {
+        if (!event)
             return;
-        }
 
-        JSON.stringify(event.params, function (key, value) {
-            $scope.selectedModulePathList.pop(value.data.id);
-        });
+        $scope.selectedModulePathList.pop(event.params.data.id);
```
Comment (mdopark) [inline project/dexter-server/public/AdminSE/adminSECtrl.js]: Not need to loop ??

File: project/dexter-server/public/AdminSE/adminSECtrl.js [hunk 10]
```diff
@@ -43,42 +43,31 @@ adminSEApp.controller('adminSECtrl', function ($scope, $http, $location, $routeP
...
     function addDidList(event) {
-        if (!event) {
+        if (!event)
             return;
-        }
 
-        JSON.stringify(event.params, function (key, value) {
-            $scope.selectedDidList.push(value.data.id);
-        });
+        $scope.selectedDidList.push(event.params.data.id);
```
Comment (mdopark) [inline project/dexter-server/public/AdminSE/adminSECtrl.js]: Not need to loop ??

File: project/dexter-server/public/AdminSE/adminSECtrl.js [hunk 11]
```diff
@@ -43,42 +43,31 @@ adminSEApp.controller('adminSECtrl', function ($scope, $http, $location, $routeP
...
     function removeDidList(event) {
-        if (!event) {
+        if (!event)
             return;
-        }
 
-        JSON.stringify(event.params, function (key, value) {
-            $scope.selectedDidList.pop(value.data.id);
-        });
+        $scope.selectedDidList.pop(event.params.data.id);
```
Comment (mdopark) [inline project/dexter-server/public/AdminSE/adminSECtrl.js]: Not need to loop ??

File: project/dexter-server/public/AdminSE/adminSECtrl.js [hunk 12]
```diff
@@ -113,17 +102,22 @@ adminSEApp.controller('adminSECtrl', function ($scope, $http, $location, $routeP
...
+            angular.forEach(result.data.rows, (index) => {
+                $scope.didList.push(index.did);
+            });
+
+            $(".did-select").select2({
+                data: $scope.didList
+            });
+
         }, function (results) {
             $log.error('Error code:' + results.status + ';');
```
Comment (mdopark) [inline project/dexter-server/public/AdminSE/adminSECtrl.js]: hide loding image...

File: project/dexter-server/public/AdminSE/adminSECtrl.js [hunk 13]
```diff
@@ -134,19 +128,24 @@ adminSEApp.controller('adminSECtrl', function ($scope, $http, $location, $routeP
...
+            });
+            $scope.modulePathList.push(NONE_MODULE_PATH);
+
+            $(".module-select").select2({
+                data: $scope.modulePathList
+            });
+
+
         }, function (results) {
             $log.error('Error code:' + results.status + ';');
```
Comment (mdopark) [inline project/dexter-server/public/AdminSE/adminSECtrl.js]: hide loading image...

**Comment by mdopark** [comment]
@Minjung-Baek  Good!!

**Pull request merged**